File: chapter6/salesitemservice/src/common/logger/StdOutLogger.ts
import { Logger, LogLevel, logLevelToSeverity } from './Logger';
import { Injectable } from '@nestjs/common';

@Injectable()
export default class StdOutLogger implements Logger {
  private readonly loggingLevelSeverity: number =
    logLevelToSeverity[process.env.LOG_LEVEL || 'WARN'] ?? 4;

  log(logLevel: LogLevel, message: string): void {
    const logLevelSeverity = logLevelToSeverity[logLevel];

    if (logLevelSeverity >= this.loggingLevelSeverity) {
      console.log(`${logLevel}: ${message}`);
    }
  }
}

File: chapter2/orderservice/src/main/java/com/example/orderservice/common/logger/StdOutLogger.java
package com.example.orderservice.common.logger;

import org.springframework.stereotype.Service;

@Service
public class StdOutLogger implements Logger {
  @Override
  public void log(LogLevel logLevel, String message) {
    System.out.println(logLevel.name() + ": " + message); // NOSONAR
  }
}
